File: spmercantil/src/main/java/github/faustofjunqueira/spmercantil/core/dto/FilterMarketDto.java
package github.faustofjunqueira.spmercantil.core.dto;

import lombok.Data;

/**
 * Informações a serem filtradas sobre as feiras
 */
@Data
public class FilterMarketDto {

    /**
     * Instacia o filtro FilterMarketDto
     *
     * @param district String Nome do Distrito Municipal
     * @param region5 String Região conforme divisão do Município em 5 áreas
     * @param name String Denominação da feira livre atribuída pela Supervisão de Abastecimento
     * @param neighborhood String Bairro de localização da feira livre
     * @param page Integer numero da pagina
     * @param size Integer Quantidade de registro na pagina
     */
    public FilterMarketDto(String district, String region5, String name, String neighborhood, Integer page, Integer size) {
        this.district = district;
        this.region5 = region5;
        this.name = name;
        this.neighborhood = neighborhood;
        this.page = page;
        this.size = size;
    }

    /**
     * Numero da pagina
     */
    private Integer page;

    /**
     * Tamanho da pagina
     */
    private Integer size;

    /**
     * Nome do Distrito Municipal
     */
    private String district;

    /**
     * Região conforme divisão do Município em 5 áreas
     */
    private String region5;

    /**
     * Denominação da feira livre atribuída pela Supervisão de Abastecimento
     */
    private String name;

    /**
     * Bairro de localização da feira livre
     */
    private String neighborhood;

}

File: SpMercantil/Core/Domain/Dto/FilterMarketDto.cs
namespace Core.Domain.Dto
{
    /// <summary>
    ///     Objeto de filtro de feira(market)
    /// </summary>
    public class FilterMarketDto
    {
        /// <summary>
        ///     Nome do Distrito Municipal
        /// </summary>
        public string? District { get; set; }

        /// <summary>
        ///     Região conforme divisão do Município em 5 áreas
        /// </summary>
        public string? Region5 { get; set; }

        /// <summary>
        ///     Denominação da feira livre atribuída pela Supervisão de Abastecimento
        /// </summary>
        public string? Name { get; set; }

        /// <summary>
        ///     Bairro de localização da feira livre
        /// </summary>
        public string? Neighborhood { get; set; }

        /// <summary>
        ///     Pagina a ser acessada. default=0
        /// </summary>
        public int Page { get; set; } = 0;

        /// <summary>
        ///     Quantidade de registros na pagina default=25.
        /// </summary>
        public int Size { get; set; } = 25;
    }
}